events/tests/TESTS/events/equeue/CMakeLists.txt

```diff
@@ -1,20 +1,19 @@
-# Copyright (c) 2020 ARM Limited. All rights reserved.
+# Copyright (c) 2021 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
+include(mbed_greentea)
 
-set(MBED_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../../.. CACHE INTERNAL "")
-set(TEST_TARGET mbed-events-equeue)
-
-include(${MBED_PATH}/tools/cmake/mbed_greentea.cmake)
-
-project(${TEST_TARGET})
+if(NOT "MBED_EXTENDED_TESTS" IN_LIST MBED_CONFIG_DEFINITIONS)
```

Comment:
If I remember correctly, the `MBED_EXTENDED_TESTS` macro was created to disable tests that are slow to run by default, to save CI time. But if a user want to enable all tests, they need to define it. How will they be able to do it?

Mbed CLI 1 has a `-D<macro>` option for this, and we don't have an equivalent in Mbed CLI 2. I can think of three options:
* add a `-D<macro>` option to mbed-tools, and generated mbed_config.cmake will contain add `<macro>` alongside other C/C++ macros which eventual get [passed to `target_compile_definitions`](https://github.com/ARMmbed/mbed-os/blob/68667df7ac76a2c2660b331a350d1a2ceaa433b7/CMakeLists.txt#L102-L106)
* add a `-D<macro>` option to mbed-tools, but `<macro>` is a CMake variable. Each test's CMakeLists.txt can check this variable
* add `TESTS/configs/extended.json` containing
```json
{
    "macros": ["MBED_EXTENDED_TESTS"]
}
```
so users can pass `--app-config TESTS/configs/extended.json` to `mbed-tools configure`.

What do you and @ARMmbed/mbed-os-core think?

Reply:
General question on mbed-tools in general, not just tests: Do we want to give users ability to set a global C/C++ macro on the fly, like Mbed CLI 1 does?

Reply:
The concept of `MBED_EXTENDED_TESTS` seems flawed to me. If we want to save time, we should do it by making tests run quicker, increasing our reliance on unit tests and de-emphasizing integration tests, not by avoiding running tests.

How about we always include `MBED_EXTENDED_TESTS` type tests when using CLI 2? This would mean no mention of `MBED_EXTENDED_TESTS` in CMake.

Reply:
> General question on mbed-tools in general, not just tests: Do we want to give users ability to set a global C/C++ macro on the fly, like Mbed CLI 1 does?

I think we should just let users run CMake directly instead of adding some pass-through for global definitions. mbed-tools is supposed to be "easy mode", if you want "advanced mode" you can use cmake directly.

Reply:
Is there a cost consideration when it comes to the time it takes to run integration tests in the CI? I was under the impression that they run overnight, so time itself wouldn't be a major concern on its own.

Reply:
> The concept of `MBED_EXTENDED_TESTS` seems flawed to me. If we want to save time, we should do it by either making tests run quicker, or by increasing our reliance on unit tests and de-emphasizing integration tests, not by avoiding running tests.
> 
> How about we always include `MBED_EXTENDED_TESTS` type tests when using CLI 2? This would mean no mention of `MBED_EXTENDED_TESTS` in CMake.

This seems reasonable to me. If the tests really take "too long" we should fix them so they run in a reasonable time.

Reply:
It might be interesting to try (e.g. on a local board) how long extended tests take, starting with events tests in this PR. We rarely use `MBED_EXTENDED_TESTS` except in a few places, so unless it really takes crazy long (which seems unlikely to me), this seems just to be inconsistency.

> The concept of MBED_EXTENDED_TESTS seems flawed to me. If we want to save time, we should do it by either making tests run quicker, or by increasing our reliance on unit tests and de-emphasizing integration tests, not by avoiding running tests.

+1

Reply:
The time saved in finding bugs or issues with a PR before it gets merged is well worth the extra time a PR job takes to run. When we are looking to make a release, we don't want to spend days tracking down why a test we never ran until the day before the scheduled release started failing: we want the failures up front and as close as possible to when the change is made, so it is easiest to fix (all context for the change is available and the person who raised the PR is still around to fix their PR rather than moved on to other things).

Reply:
From elapsed times reported by mbedgt:
- `events/equeue: 29.33 sec` (currently skipped w/o MBED_EXTENDED_TESTS defined)
- `fat_filesystem: 21.25 sec` (currently skipped w/o MBED_EXTENDED_TESTS defined)
- `multipart_fat_filesystem: 23.16 sec` (currently skipped w/o MBED_EXTENDED_TESTS defined)

For comparison: 
- `events/queue: 29.42` (not skipped
- `events/timing: 80.12 sec` (not skipped)

Do we think completely removing `MBED_EXTENDED_TESTS` from the codebase is the best way forward?

Reply:
> Do we think completely removing MBED_EXTENDED_TESTS from the codebase is the best way forward?

Yep. A few minutes of testing time is well worth it.

Reply:
Rebased with `MBED_EXTENDED_TESTS` removal commit.